O/onnxruntime/build_tarballs.jl

```diff
@@ -0,0 +1,83 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "onnxruntime"
+version = v"1.10.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/microsoft/onnxruntime.git", "0d9030e79888d1d5828730b254fedc53c7b640c1"),
+    ArchiveSource("https://github.com/protocolbuffers/protobuf/releases/download/v3.16.1/protoc-3.16.1-linux-x86_64.zip", "dffb7209d31b7e87e8e8ba2d5d869ceda5a5cea8883c4b13a726611a0dbd8a7c"; unpack_target = "protoc"),
+    ArchiveSource("https://github.com/microsoft/onnxruntime/releases/download/v$version/onnxruntime-win-x64-$version.zip", "a0c6db3cff65bd282f6ba4a57789e619c27e55203321aa08c023019fe9da50d7"),
+    ArchiveSource("https://github.com/microsoft/onnxruntime/releases/download/v$version/onnxruntime-win-x86-$version.zip", "fd1680fa7248ec334efc2564086e9c5e0d6db78337b55ec32e7b666164bdb88c")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+
+if [[ $target == *-w64-mingw32* ]]; then
+    if [[ $target == *-w64-mingw32* ]]; then
+        if [[ $target == x86_64-w64-mingw32* ]]; then
+            dist_name=onnxruntime-win-x64
+        elif [[ $target == i686-w64-mingw32* ]]; then
+            dist_name=onnxruntime-win-x86
+        fi
+        chmod 755 $dist_name*/lib/*
+    fi
+
+    mkdir -p $includedir $libdir
+    cp -a $dist_name*/include/* $includedir
+    cp -a $dist_name*/lib/* $libdir
+    install_license $dist_name*/LICENSE
+else
+    if [[ $target == aarch64-apple-darwin* ]]; then
+        cmake_extra_defines=CMAKE_OSX_ARCHITECTURES='arm64'
```

Comment:
We already set [`CMAKE_SYSTEM_PROCESSOR`](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html), is that not enough?  Also, `-arch` is already set correctly:
```console
sandbox:${WORKSPACE} # echo 'int main(){}' |  cc -x c - -###
clang version 12.0.0 (/home/mose/.julia/dev/BinaryBuilderBase/deps/downloads/llvm-project.git d28af7c654d8db0b68c175db5ce212d74fb5e9bc)
Target: arm64-apple-darwin20
Thread model: posix
InstalledDir: /opt/x86_64-linux-musl/bin
 "/opt/x86_64-linux-musl/bin/clang-12" "-cc1" "-triple" "arm64-apple-macosx11.0.0" "-Wundef-prefix=TARGET_OS_" "-Werror=undef-prefix" "-Wdeprecated-objc-isa-usage" "-Werror=deprecated-objc-isa-usage" "-emit-obj" "-mrelax-all" "--mrelax-relocations" "-disable-free" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "-" "-mrelocation-model" "pic" "-pic-level" "2" "-mframe-pointer=non-leaf" "-fno-rounding-math" "-munwind-tables" "-fcompatibility-qualified-id-block-type-checking" "-fvisibility-inlines-hidden-static-local-var" "-target-cpu" "apple-a12" "-target-feature" "+neon" "-target-abi" "darwinpcs" "-fallow-half-arguments-and-returns" "-debugger-tuning=lldb" "-nostdinc++" "-resource-dir" "/opt/x86_64-linux-musl/lib/clang/12.0.0" "-isystem" "/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/usr/include/c++/v1" "-isysroot" "/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root" "-internal-isystem" "/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/usr/local/include" "-internal-isystem" "/opt/x86_64-linux-musl/lib/clang/12.0.0/include" "-internal-externc-isystem" "/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/usr/include" "-Wno-unused-command-line-argument" "-fdebug-compilation-dir" "/workspace" "-ferror-limit" "19" "-stack-protector" "1" "-fblocks" "-fencode-extended-block-signature" "-fregister-global-dtors-with-atexit" "-fgnuc-version=4.2.1" "-fmax-type-align=16" "-fcolor-diagnostics" "-o" "/tmp/--3fffb7.o" "-x" "c" "-"
 "/opt/bin/aarch64-apple-darwin20-libgfortran5-cxx11/ld64.aarch64-apple-darwin20" "-dynamic" "-arch" "arm64" "-headerpad_max_install_names" "-macosx_version_min" "11.0.0" "-syslibroot" "/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root" "-o" "a.out" "-L/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/lib" "/tmp/--3fffb7.o" "-lSystem"
```

Should this variable be really necessary (and I'm not yet convinced of that), we should rather put it in [`BuildToolchains.jl`](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/e7286d0d7e0c1b7d724d173529edd12ecb411eba/src/BuildToolchains.jl)

Reply:
Did you try without this?

Reply:
It fails to link https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=16920&view=logs&j=e9df152d-f6ae-5301-9d40-c17cf2618feb&t=976e9440-1181-57d1-e944-a3ddf8007e43&l=1892

Reply:
Ok, I suspect they're doing something dodgy with `CMAKE_OSX_ARCHITECTURES`, like here https://github.com/microsoft/onnxruntime/blob/481b96d32aecc5fceabc9e9401604d2d31212ef7/tools/ci_build/build.py#L1952-L1953 where they use `platform.machine()` which would use the information about the build machine

Reply:
Cf. https://github.com/microsoft/onnxruntime/blob/v1.10.0/cmake/CMakeLists.txt#L186